### PR TITLE
[Proposal of testing approach] Write tests for issue API

### PR DESCRIPTION
**Pull request by andreynering**
Context: https://github.com/gogits/gogs/issues/2293

I wrote few tests (just 4), because I want to know if you guys agree with the general approach. More could be easily added for other APIs (or existing tests improved).

### Approach

- We are now testing against a SQLite database (It could be PostgreSQL, or MySQL. I choose SQLite for portability)
- Fixtures are loaded in the test database before each router test
- In the future, we might need to write scripts that generate Git repositories in the `gogs-repositories` folder, in order to test code that traverse Git repos.

### CODE REVIEW

File: modules/setting/setting.go [hunk 1]
```diff
@@ -33,6 +33,9 @@ const (
 	HTTP  Scheme = "http"
 	HTTPS Scheme = "https"
 	FCGI  Scheme = "fcgi"
+
+	RUN_MODE_PROD = "prod"
+	RUN_MODE_TEST = "test"
 )
```
Comment (mhartkorn) [inline modules/setting/setting.go]: Did you leave out `RUN_MODE_DEV` on purpose?

Comment (andreynering) [inline modules/setting/setting.go]: I leave it out because it wouldn't be used by now. But I think I could add it for maybe future use.

**Comment by andreynering** [comment]
Tip: tests compiles a lot faster without `-cover` and `-race` flags. You probably won't want to use that for development. I took many minutes to discover why it was so slow.

**Comment by andreynering** [comment]
@Unknwon Any feedback here?

**Comment by Unknwon** [comment]
I think it's not a good time to spend time on integrating testing yet.

**Comment by andreynering** [comment]
@Unknwon OK. I hope Gogs get a strong test suite in the future, even that a different approach would be choose.

**Comment by Unknwon** [comment]
@andreynering I think Gogs should have as much as unit tests as possible, combine with integrating tests. But thoughts change fast, so I don't want people keep putting time on testing.

**Pull request closed without merging**